ui/pages/routes/QRHardwareStateWatcher.js

```diff
@@ -0,0 +1,18 @@
+import React, { useCallback, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import * as actions from '../../store/actions';
+import { getCurrentQRHardwareState } from '../../selectors';
+
+export default () => {
```

Comment:
I might be misreading code, but I think what this is trying to achieve can be more easily done (and done in a way in line with our recent patterns) but using our `Popover` component instead of the modal component, selecting the `qrHardware` state in the appropriate parent of the popover, and displaying conditional on that state.

Look at `RecoveryPhraseReminder` in `home.component.js` for an example

- `RecoveryPhraseReminder` uses the `Popover` component
- we get `showRecoveryPhraseReminder` from state
- we show the `RecoveryPhraseReminder` component if `showRecoveryPhraseReminder` is true

This way, you don't need a component to watch for background changes through a use effect, and then dispatch a new action in response.

Reply:
Let me know if you have further questions on this.

Reply:
Hi @danjm , Thanks for your review. Absolutely the `Popover` is a good pattern in this situation. 
The reason we choose `Modal` is that in the current signing process, after user click `confirm` in `send page` the route will go back to `home page` and all previous components will unmount. The `Modal` component will not be affected by this process. 
So we have two options: 
1. We still use `Modal` component to do wallet import and perform signing request.
2. We use a root level `Popover` to do the jobs. 

We'd prefer option 2, but I did not find a root level `Popover` in the code, not sure is that a good pattern. 
What do you think of?

Reply:
For option2 We can use `Popover` to refactor the `QRHardwareStateWatcher`(name can be changed) which will be the wrapper component of all QR Reader or Player

Reply:
yeah, I think @aaronisme is right here: you can put a new component, which is the same as the current modal component, but wrapped in a `Popover` instead, in the same place that the `QRHardwareWatcher` currently is. So I think it is fine to have a component like that in `routes.component.js`